chore(*): refactor, improve, upgrade

gkalpak opened the pull request:
Main themes:
- Improve file/directory layout.
- Switch from bower to npm.
- Upgrade all dependencies.

Incorporates #361 and #390.
Partially addresses #334.

gkalpak commented:
Not sure why Travis isn't triggered. I verified this on my fork: https://travis-ci.org/gkalpak/angular-seed/builds/446783519

petebacondarwin commented on README.md:
```diff
@@ -3,7 +3,7 @@
 to quickly bootstrap your angular webapp projects and dev environment for these projects.
 
-The seed contains a sample AngularJS application and is preconfigured to install the Angular
+The seed contains a sample AngularJS application and is preconfigured to install the AngularJS
```
:+1:

petebacondarwin commented on README.md:
```diff
@@ -39,30 +39,30 @@ The `depth=1` tells git to only pull down one commit worth of historical data.
-Behind the scenes this will also call `bower install`. After that, you should find out that you have
-two new folders in your project.
+Behind the scenes this will also call `npm run copy-libs`, which copies the AngularJS files and
+other frontend dependencies. After that, you should find out that you have two new directories in
```
front end is two words (possibly hyphenated).

mgol commented:
> why rename the folder `components` to `core`?

@petebacondarwin See #334.

gkalpak commented on README.md:
````diff
@@ -236,7 +236,7 @@ choose to install the tool globally:
 ```
 
-Then you can start your own development web server to serve static files from a folder by running:
+Then you can start your own development web server to serve static files from any folder by running:
````
How about `unknown`? 😛

gkalpak commented on .travis.yml:
```diff
@@ -1,21 +1,23 @@
-  - npm install
+before_install:
+  - export DISPLAY=":99.0";
+  - sh -e /etc/init.d/xvfb start;
```
I was a little on the fence on that one. (Might be a little too magic for beginners.)
But I am not opposed. Let's give it a go.

gkalpak commented:
- Addressed comments.
- Added a couple of `fixup` commits and also [switched to Chrome headless on CI][1].
- Ran Travis [on my fork][2].

@petebacondarwin, @mgol: PTAL

[1]: https://github.com/angular/angular-seed/pull/444/commits/a02c9f2c11393c6943fc4d233199b6bf6f34cb1a
[2]: https://travis-ci.org/gkalpak/angular-seed/builds/448246204

Narretz commented:
Looks like Travis is not enabled for this repo - it must have been disabled some time in the last year: https://travis-ci.org/angular/angular-seed (says "This is not an active repository" for me)

petebacondarwin commented:
And we don't have rights to turn it back on...
